tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs

```diff
@@ -1401,6 +1410,16 @@ private async Task CoordinateTasks(string loginId, Task<RepositorySearchInformat
             SelectionOptionsPlaceholderText = repoSearchInformation.SelectionOptionsPlaceHolderText;
 
             IsFetchingRepos = false;
+
+            if (!_repositoriesForAccount.Any())
+            {
+                NoRepositoriesMessage = _stringResource.GetLocalized(StringResourceKey.RepoToolNoRepositoriesMessage, _providers.DisplayName(_selectedRepoProvider));
+                ShowRepoPage = false;
+                ShouldShowNoRepoMessage = true;
+                ShouldShowGranularSearch = false;
+                FolderPickerViewModel.ShouldShowFolderPicker = false;
+                EditDevDriveViewModel.ShowDevDriveInformation = false;

```

Comment:
Design Q: Seems like you're building a state machine where each state is a set of boolean values { false, true, false, false, false}.

Seems like it would be better to have have a proper state machine, or a user control for each state?

(my instinct says that "ShowRepoPage = !ShouldShowNoRepoMessage" so that you don't need 2 vars, but all the other stuff in this view complicates that)

Reply:
You are absolutely correct.  DevHome already does this in other parts.  I think "Main Page"  However, I made this when I had no idea templates was a thing.  Moving the UI to a template system, would work much better.